src/utils.jl

```diff
@@ -0,0 +1,129 @@
+# generating graphs
+function square_lattice(mx::Int, my::Int, p::Float64; seed::Int = 1234)
+    Random.seed!(seed)
+    n = Int(ceil(mx * my * p))
+    g = SimpleGraph(n)
+    positions = [(i, j) for i in 1:mx, j in 1:my]
+    shuffle!(positions)
+    positions = positions[1:n]
+    sort!(positions)
+    for (i, pos) in enumerate(positions)
+        mxi, myi = pos
+        for (dx, dy) in [(0, 1), (1, 0)]
+            mxj, myj = mxi + dx, myi + dy
+            if (mxj, myj) in positions
+                j = findfirst(x -> x == (mxj, myj), positions)
+                add_edge!(g, i, j)
+            end
+        end
+    end
+    return g
+end
+
+function chain(n::Int)
+    g = SimpleGraph(n)
+    for i in 1:n-1
+        add_edge!(g, i, i+1)
+    end
+    return g
+end
+
+# codes about constructing the einsum expression
+
+function all_eins(g::SimpleGraph{Int}, count::Int)
```

Comment:
Bad design. Better not use count, which is a contextual information, as an argument. When I have a similar need of generating new indices, I usually pass an index store object to this function, e.g. https://github.com/CodingThrust/SimpleTDVP.jl/blob/fb946b589dc094b02587154d96efc8bc1ea99ec6/src/utils.jl#L1